src/main/java/dynamicfps/DynamicFPSMod.java

```diff
@@ -104,44 +144,56 @@ private static void checkForStateChanges() {
 			}
 		}
 	}
-	
+
 	private static void onFocus() {
 		setVolumeMultiplier(1);
+		GameOptions gameOptions = MinecraftClient.getInstance().options;
+		increaseGraphicsFully(gameOptions);
+		increaseGraphics(gameOptions);
```

Comment:
These methods can not be called unconditionally, as all of the `previousX` fields have not necessarily been initialized (or rather, are the default, so eg. boolean values are `false`) when the options are disabled, causing these options to be set to erroneous values.
```suggestion
		if (config.fullyReduceGraphicsWhenUnfocused) {
			increaseGraphicsFully(gameOptions);
		}
		if (config.reduceGraphicsWhenUnfocused) {
			increaseGraphics(gameOptions);
		}
```